File: students/students/src/MON.SPA/src/models/reassessmentModel.js
import { DocumentModel } from './documentModel.js';
export class ReassessmentModel {
  constructor(obj = {}) {
    this.id = obj.id,
    this.personId = obj.personId;
    this.reasonId = obj.reasonId;
    this.reason = obj.reason;
    this.reasonName = obj.reasonName;
    this.inClass = obj.inClass || "";
    this.basicClassName = obj.basicClassName;
    this.reassessmentDetails = obj.reassessmentDetails || [];
    this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
    this.sortOrder = obj.sortOrder;
    this.schoolYear = obj.schoolYear;
    this.schoolYearName = obj.schoolYearName;
  }
}

File: students/students/src/MON.Models/ReassessmentModel.cs
namespace MON.Models
{
    using System;
    using System.Collections.Generic;

    public class ReassessmentModel
    {
        public int Id { get; set; }

        public int PersonId { get; set; }

        public int ReasonId { get; set; }
        public string ReasonName { get; set; }

        public int? InClass { get; set; }

        public short? SchoolYear { get; set; }
        public string SchoolYearName { get; set; }

        public string Reason { get; set; }

        public string BasicClassName { get; set; }

        public DateTime CreationDate { get; set; }
        public IEnumerable<ReassessmentDetailsModel> ReassessmentDetails { get; set; }
        public IEnumerable<DocumentViewModel> Documents { get; set; }

    }
}
